### PR TITLE
Detect layers by looking at header.wkw

### LINKED ISSUE #148: Detect layers by looking for header.wkw
When a layer is not named `color`, `segmentation` or `prediction` it is recognized as a wkw layer. However, it should be possible to detect a layer simply by looking for a `header.wkw` file. Then we can detect arbitrary layer names.

### PR DESCRIPTION
**Pull request by rschwanhold**
closes #148

### CODE REVIEW

File: wkcuber/metadata.py [hunk 1]
```diff
@@ -12,6 +12,8 @@
 from .mag import Mag
 from typing import List
 from .utils import add_verbose_flag, setup_logging, add_scale_flag
+from pathlib import Path
+from os.path import *
```
Comment (philippotto) [inline wkcuber/metadata.py]: I consider `import *` as an anti-pattern as it hides the fact where functions are taken from. For example, the code itself doesn't really tell me that `glob` stems from `os.path`. Additionally, there is the risk that this wildcard import shadows other variables in this namespace. Therefore, I'd advise to state explictly what you want to import. Alternatively, you can just do `import os.path` and then use glob like `os.path.glob`, I think.

File: wkcuber/metadata.py [hunk 2]
```diff
@@ -309,13 +311,22 @@ def detect_segmentation_layer(
...
             dataset_path, "segmentation", max_id, compute_max_id, exact_bounding_box
         )
+    available_layer_names = set(
+        [
+            basename(normpath(Path(x).parent.parent))
+            for x in glob.glob(dataset_path + "/*/*/header.wkw")
+        ]
+    )
+    for layer_name in available_layer_names:
+        # color and segmentation are already checked explicitly to ensure downwards compatibility
```
Comment (philippotto) [inline wkcuber/metadata.py]:
```suggestion
        # color and segmentation are already checked explicitly to ensure downwards compatibility (some older datasets don't have the header.wkw file)
```

File: wkcuber/metadata.py (hunk 1 above)
Comment (rschwanhold) [reply wkcuber/metadata.py]: I see what you mean. Actually glob is not from `os.path` (it was already imported).

Comment (philippotto) [reply wkcuber/metadata.py]: > Actually glob is not from `os.path` (it was already imported).

that drives the point home, too :laughing:

Comment (hotzenklotz) [reply wkcuber/metadata.py]: #advertisment Me likey `pathlib.rglob` https://docs.python.org/3/library/pathlib.html#pathlib.Path.rglob

**Pull request merged**